trainer/lib/trainer/xcresult.rb

```diff
@@ -184,16 +186,10 @@ def find_failure(failures)
           #     or identifier: "TestThisDude/testFailureJosh2" (when Objective-C)
 
           found_failure = failures.find do |failure|
-            # Clean test_case_name to match identifier format
-            # Sanitize for Swift by replacing "." for "/"
-            # Sanitize for Objective-C by removing "-", "[", "]", and replacing " " for ?/
-            sanitized_test_case_name = failure.test_case_name
-                                              .tr(".", "/")
-                                              .tr("-", "")
-                                              .tr("[", "")
-                                              .tr("]", "")
-                                              .tr(" ", "/")
-            self.identifier == sanitized_test_case_name
+            # Sanitize both test case name and identifier in a consistent fashion, then replace all non-word
+            # chars with underscore, and compare them
+            sanitized_test_case_name = sanitizer.call(failure.test_case_name)
+            sanitized_identifier == sanitized_test_case_name
```

Comment:
This change is a regression. See https://github.com/fastlane/fastlane/pull/21432#issuecomment-2067310932

Reply:
Fix https://github.com/fastlane/fastlane/pull/21990